packages/generator/src/v1.1/generateActions.ts

```diff
@@ -0,0 +1,133 @@
+  ontology: WireOntologyDefinition,
+  fs: MinimalFs,
+  outDir: string,
+) {
+  const importedObjects = new Set<string>();
+  let actionSignatures = [];
+  for (const action of ontology.actionTypes) {
+    const entries = Object.entries(action.parameters);
+
+    const modifiedEntityTypes = getModifiedEntityTypes(action);
+    const addedObjects = Array.from(modifiedEntityTypes.addedObjects);
+    const modifiedObjects = Array.from(modifiedEntityTypes.modifiedObjects);
+    addedObjects.forEach(importedObjects.add, importedObjects);
+    modifiedObjects.forEach(importedObjects.add, importedObjects);
+
+    let jsDocBlock = ["/**"];
+    if (action.description) {
+      jsDocBlock.push(`* ${action.description}`);
+    }
+
+    let parameterBlock = "";
+    if (entries.length > 0) {
+      parameterBlock = `params: { \n`;
+      for (
+        const [parameterName, parameterData] of entries
+      ) {
+        parameterBlock += `"${parameterName}"`;
+        parameterBlock += parameterData.required ? ": " : "?: ";
+        const getTypeScriptType = handleDataType(
+          parameterData.dataType,
+          importedObjects,
+        );
+        parameterBlock += `${getTypeScriptType};\n`;
+
+        jsDocBlock.push(
+          `* @param {${getTypeScriptType}} params.${parameterName}`,
+        );
+      }
+      parameterBlock += "}, ";
+    }
+
+    jsDocBlock.push(`*/`);
+    actionSignatures.push(
+      `
+      ${jsDocBlock.join("\n")}
+      ${action.apiName}<O extends ActionExecutionOptions>(${parameterBlock}options?: O): 
+        Promise<Result<ActionResponseFromOptions<O, Edits<${
+        addedObjects.length > 0
+          ? addedObjects.join(" | ")
+          : "void"
+      }, ${
+        modifiedObjects.length > 0
+          ? modifiedObjects.join(" | ")
+          : "void"
+      }>>, ActionError>>;`,
+    );
+  }
+
+  await fs.writeFile(
+    `${outDir}/ontologyActions.ts`,
+    await formatTs(`
+    import type { ObjectSet, LocalDate, Timestamp, Attachment, Edits, ActionExecutionOptions, ActionError, Result, ActionResponseFromOptions } from "@osdk/legacy-client";
+    ${
+      Array.from(importedObjects).map(importedObject =>
+        `import type { ${importedObject} } from "./objects/${importedObject}";`
+      ).join("\n")
+    }
+    export interface Actions {
+    ${actionSignatures.join("\n")}
+    }
+  `),
+  );
+}
+
+function handleDataType(
+  actionParameter: ActionParameterType,
+  importedObjects: Set<string>,
+): string {
+  switch (actionParameter.type) {
+    case "objectSet": {
+      const objectType = actionParameter.objectTypeApiName!;
+      importedObjects.add(objectType);
+      return `ObjectSet<${objectType}>`;
+    }
+    case "object": {
+      const objectType = actionParameter.objectTypeApiName!;
+      importedObjects.add(objectType);
+      return `${objectType}`;
+    }
+    case "array":
+      return handleDataType(actionParameter.subType, importedObjects) + `[]`;
```

Comment:
I did this as `Array<${innerType}>` just in case I got a union, the `[]` would be sort of ambiguous. I don't think you have that problem here though.